packages/context/src/tests/signalr/useSignalRConnection.unit.tsx

```diff
@@ -0,0 +1,172 @@
+        );
+
+        sinon.verify();
+    });
+
+    it('notifies callback of instance changes', () => {
+        let callback: Parameters<InstanceSubscription['subscribe']>[1] | undefined;
+
+        sinon.stub(instanceSubscription, 'subscribe').callsFake((_, cb) => {
+            callback = cb;
+        });
+
+        sinon.mock(callbacks).expects('instanceCallback').withExactArgs('instance1', 'instance2');
+
+        instanceTopic = 'testObject';
+
+        render(
+            <NotificationContext.Provider value={notifications}>
+                <SignalRConnectionProvider>
+                    <TestComponent />
+                </SignalRConnectionProvider>
+            </NotificationContext.Provider>,
+        );
+
+        expect(callback).to.not.be.undefined();
+
+        // Simulate a notification.
+        callback?.(
+            { objectId: 'testObject', instanceId: 'instance1' },
+            { objectId: 'testObject', instanceId: 'instance2' },
+        );
+
+        sinon.verify();
+    });
+
+    it('subscribes to document change notifications', () => {
+        sinon.mock(documentSubscription).expects('subscribe').withArgs('docObject', 'testInstance');
+
+        documentTopic = 'docObject/testInstance';
+
+        render(
+            <NotificationContext.Provider value={notifications}>
+                <SignalRConnectionProvider>
+                    <TestComponent />
+                </SignalRConnectionProvider>
+            </NotificationContext.Provider>,
+        );
+
+        sinon.verify();
+    });
+
+    it('notifies callback of document changes', () => {
+        let callback: Parameters<DocumentSubscription['subscribe']>[2] | undefined;
+
+        sinon.stub(documentSubscription, 'subscribe').callsFake((_, _1, cb) => {
+            callback = cb;
+        });
+
+        sinon.mock(callbacks).expects('documentCallback').withExactArgs(
+            {
+                objectId: 'docObject',
+                instanceId: 'testInstance',
+                documentId: 'doc1',
+                type: 'type1',
+            },
+            {
+                objectId: 'docObject',
+                instanceId: 'testInstance',
+                documentId: 'doc2',
+                type: 'type2',
+            },
+        );
+
+        documentTopic = 'docObject/testInstance';
+
+        render(
+            <NotificationContext.Provider value={notifications}>
+                <SignalRConnectionProvider>
+                    <TestComponent />
+                </SignalRConnectionProvider>
+            </NotificationContext.Provider>,
+        );
+
+        expect(callback).to.not.be.undefined();
+
+        // Simulate a notification.
+        callback?.(
+            { objectId: 'docObject', instanceId: 'testInstance', documentId: 'doc1', type: 'type1' },
+            { objectId: 'docObject', instanceId: 'testInstance', documentId: 'doc2', type: 'type2' },
+        );
+
+        sinon.verify();
+    });
+});
```

Comment:
@actong do we also want to test if `unsubscribe` methods are being called properly? Also I don't know how in-depth we want to go, but maybe also test for using the same instance of the callback function?

Reply:
Hmm, yeah, good point, right now it wouldn't work for instances since it's always a new function.

Reply:
I wonder if there's a way to still use useNotification with instances here without impact backwards compatibility.